specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,557 @@
+  /// The Texture stream becomes 'Started' state once it starts sending a texture
+  /// until it calls Stop API or receives 'StopRequested' event.
+  HRESULT add_StartRequested(
+      [in] ICoreWebView2StagingTextureStreamStartRequestedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for start stream request.
+  HRESULT remove_StartRequested(
+      [in] EventRegistrationToken token);
+  /// Listen to stop stream request once the stream started.
+  /// It is called when user stop all streaming requests from
+  /// the renderers (Javascript) or the host calls the Stop API. The renderer
+  /// can stream again by calling the streaming request API.
+  /// The renderer cleared all registered TextureBuffers before sending
+  /// the stop request event so that the callback of the next start request
+  /// should register the textures again.
+  /// The event is triggered when all requests for given stream id closed
+  /// by the Javascript, or the host's Stop API call.
+  /// TextureBuffer related API calls after this event will return an error
+  /// of HRESULT_FROM_WIN32(ERROR_INVALID_STATE).
+  HRESULT add_StopRequested(
+      [in] ICoreWebView2StagingTextureStreamStopRequestedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for stop stream request.
+  HRESULT remove_StopRequested(
+      [in] EventRegistrationToken token);
+  /// Creates TextureBuffer that will be referenced by the host and the browser.
+  /// By using the TextureBuffer mechanism, the host does not have to
+  /// send respective texture to the renderer, instead it notifies it
+  /// with internal TextureBuffer id, which is the identity of the TextureBuffer.
+  /// The TextureBuffer is 2D texture, IDXGIResource, format and will be
+  /// exposed through shared HANDLE or IUnknown type through ICoreWebView2StagingTexture.
+  /// Whenever the host has new texture to write, it should ask
+  /// reusable ICoreWebView2StagingTexture from the GetAvailableBuffer,
+  /// which returns ICoreWebView2StagingTexture.
+  /// If the GetAvailableBuffer returns an error, then the host calls the
+  /// CreateBuffer to allocate new TextureBuffer.
+  /// The API also registers created shared handle to the browser once it
+  /// created the resource.
+
+  /// Unit for width and height is texture unit (in texels).
+  /// 'https://learn.microsoft.com/en-us/windows/win32/api/d3d11ns-d3d11-d3d11_texture2d_desc'
+  HRESULT CreateBuffer(
+    [in] UINT32 width,
+    [in] UINT32 height,
+    [out, retval] ICoreWebView2StagingTexture** buffer);
+  /// Returns reuseable TextureBuffer for video frame rendering.
+  /// Once the renderer finishes rendering of TextureBuffer's video frame, which
+  /// was requested by Present, the renderer informs the host so that it can
+  /// be reused. The host has to create new TextureBuffer with CreateBuffer
+  /// if the API return an error HRESULT_FROM_WIN32(ERROR_NO_MORE_ITEMS).

```

Comment:
It is up to the system, usually from 3 to 5 buffers. If the host sends image (call Present) with 120 fps, but the renderer is slow, then it could go more than 10 buffers.   If the host can't create more buffers, then it can remove buffer with RemoveBuffer of old frame even though it might not fully be rendered yet in the browser. This is a mechanism already used in the Chromium engine for camera capture.